File: composeApp/src/commonMain/kotlin/com/district37/toastmasters/models/Event.kt
package com.district37.toastmasters.models

/**
 * Represents an individual detailed event
 */
data class Event(
    val id: Int,
    val images: List<String>,
    val title: String,
    val description: String,
    val time: Long,
    val locationInfo: String,
    val agenda: List<AgendaItem>,
    val additionalLinks: List<ExternalLink>,
)

data class EventPreview(
    val id: Int,
    val image: String,
    val title: String,
    val time: Long,
    val locationInfo: String,
)

data class AgendaItem(
    val title: String,
    val description: String,
    val time: Long,
    val locationInfo: String
)

data class ExternalLink(
    val displayName: String,
    val url: String
)

File: event-manager/src/types/Event.ts
export interface TimeRange {
  startTime: number;
  endTime: number;
}

export interface AgendaItem {
  title: string;
  description: string;
  time: TimeRange;
  locationInfo: string;
}

export interface ExternalLink {
  displayName: string;
  url: string;
}

export enum EventTag {
  NORMAL = 'NORMAL',
  HIGHLIGHTED = 'HIGHLIGHTED',
  BREAK = 'BREAK'
}

export interface Event {
  id: number;
  title: string;
  description: string;
  images: string[];
  time: TimeRange;
  locationInfo: string;
  agenda: AgendaItem[];
  additionalLinks: ExternalLink[];
  dateKey: string;
  tag: EventTag;
}

export interface EventPreview {
  id: number;
  title: string;
  image?: string;
  time: string;
  locationInfo: string;
}

// Helper function to format epoch timestamp to local time string
export const formatEventTime = (timestamp: number): string => {
  const date = new Date(timestamp);
  const timeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
  return date.toLocaleString('en-US', {
    weekday: 'long',
    month: 'long',
    day: 'numeric',
    hour: 'numeric',
    minute: '2-digit',
    hour12: true,
    timeZone
  });
};

// Helper function to format time range
export const formatTimeRange = (startTime: number, endTime: number): string => {
  const timeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
  const start = new Date(startTime);
  const end = new Date(endTime);
  
  const startStr = start.toLocaleString('en-US', {
    hour: 'numeric',
    minute: '2-digit',
    hour12: true,
    timeZone
  });
  
  const endStr = end.toLocaleString('en-US', {
    hour: 'numeric',
    minute: '2-digit',
    hour12: true,
    timeZone
  });
  
  return `${startStr} - ${endStr}`;
}; 